File: problem19/p19.py
def is_leap_year(year):
    if year % 4 == 0:
        if year % 1000 == 0:
            if year % 400 == 0:
                return True
            else:
                return False
        else:
            return True
    else:
        return False


def days_of_month(month, year):
    days = [0, 31, 0, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    if month != 2:
        return days[month]
    else:
        if is_leap_year(year):
            return 29
        else:
            return 28


def counting_sundays(from_year, to_year):
    current_day = 1
    count = 0

    for i in range(1900, to_year + 1):
        for j in range(1, 13):
            if current_day == 7:
                if i >= from_year:
                    count += 1
            days = days_of_month(j, i)
            if days == 29:
                current_day += 1
            if days == 30:
                current_day += 2
            if days == 31:
                current_day += 3
            if current_day > 7:
                current_day %= 7

    return count


print("Result is {}".format(counting_sundays(1901, 2000)))

File: problem19/p19.java

	    else
	        return false;
	}
	
	public static int days_of_month(int month, int year){
	    int[] days = {0, 31, 0, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
	    if(month != 2)
	        return days[month];
	    else{
	        if(is_leap_year(year))
	            return 29;
	        else{
	            return 28;
	        }
	    }
	}
	
	public static int counting_sundays(int from, int to){
	    int i, j, currentDay = 1, count = 0;

	    for(i=1900; i<=to; i++){
	        for(j=1; j<=12; j++){
	            if(currentDay == 7){
	                if(i >= from)
	                    count += 1;
	            }
	            switch(days_of_month(j, i)){
	                case 29:
	                    currentDay += 1;
	                    break;
	                case 30:
	                    currentDay += 2;
	                    break;
	                case 31:
	                    currentDay += 3;
	                    break;
	            }
	            if(currentDay > 7){
	                currentDay = currentDay % 7;
	            }
	        }
	    }

	    return count;
	}

	public static void main(String args[]) {
		
		System.out.println("Result is " + counting_sundays(1901, 2000));
		
	}
	
}
